SAGE Journals.js

```diff
@@ -116,36 +117,35 @@ function scrape(doc, url) {
 					item.title += ': ' + subtitle.trim();
 				}
 			}
-			// The encoding of apostrophs in the RIS are incorrect and
-			// therefore we extract the abstract again from the website.
+			
+			// ubtue: extract translated and other abstracts from the different xpath
+			var ubtueabstract = ZU.xpathText(doc, '//article//div[contains(@class, "abstractSection")]/p');
+			var otherabstract = ZU.xpathText(doc, '//article//div[contains(@class, "tabs-translated-abstract")]/p');
 			var abstract = ZU.xpathText(doc, '//article//div[contains(@class, "abstractSection")]/p');
-			if (abstract) {
-				item.abstractNote = abstract.trim();
-			}
-
-			// ubtue: also add translated abstracts in 520
-			var translatedAbstract = ZU.xpathText(doc, '//article//div[contains(@class, "tabs-translated-abstract")]/p');
-			if (translatedAbstract) {
+			if (ubtueabstract && otherabstract) {
+				item.abstractNote = ubtueabstract;
 				item.notes.push({
-					note: "abs:" + translatedAbstract,
+					note: "abs:" + otherabstract.replace(/^Résumé/, ''),
 				});
+			} else if (ubtueabstract && !otherabstract) {
```

Comment:
Why is `ubtueabtract` overwritten although it exists?

Reply:
to avoid the repetition of abstracts, you must first use these two xpath https://github.com/socheres/translators/blob/bb621989f091be6136bb8944834c096876e098cb/SAGE%20Journals.js#L122-L123 and if it does not exist, then overwrite it with the new xpath. E.g. here
https://github.com/ubtue/DatenProbleme/issues/1236

Reply:
But the if-condition `ubtuestract` is taken if it exists. In general, for future reference it might be useful to add a comment or reference to such issues in the source file.